Add Items to Tech Pt.2 and Cleanup

kjbranch opened the pull request:
This is very similar to previous PRs on the topic, but also with a bit of general helper cleanup.

Note that I removed the extension/requirement of `canXRayCeilingClip` in `canPreciseCeilingClip`. Although they are progressive in difficulty and accomplish the same thing, they didn't really feel like extensions to me. Let me know if you think it should be added back.

osse101 commented on tech.json:
```diff
@@ -1808,33 +1809,32 @@
               "techRequires": [
                 "canCeilingClip",
-                "canUseEnemies"
+                "canUseEnemies",
```
There shouldn't be a `canUseEnemies` requirement here since there are a number of places where you can unmorph in a two tile gap ... was it brought over with the trickyIceClip refactor?

osse101 replied:
Or maybe this tech is more enemy oriented than the name suggests

kjbranch replied:
I was talking to Maddo about this, and we could move `canUseEnemies` into the `otherRequires` and still be able to reference it without. Alternatively I was thinking we could keep `canUseEnemies` off of it and add it on with a helper, such as `h_canXRayMorphIceClip` or something?

osse101 replied:
`canXRayCeilingClip` has 2 uses: on top of enemies and between solid tiles like in Beta PBs.  Then another one is at the top of an xray climb as seen in Bowling to avoid taking spike damage, but that is less the focus of the strat.

Helpers for both sounds good to me.  And this is used for snail clips as well as ice clips.

osse101 commented on region/brinstar/red/Beta Power Bomb Room.json:
```diff
@@ -400,7 +400,7 @@
         "canTrickyJump",
         {"or": [
-          "h_canXRayCeilingClip",
+          "canXRayCeilingClip",
```
The usage here is
```
"canXRayStandUp",
"canPartialFloorClip",
"canCeilingClip"
```

kjbranch commented:
Removed `canUseEnemies` from `canXRayCeilingClip` and `canPreciseCeilingClip`.
Created two new helpers: `h_canXRayMorphIceClip`, `h_canPreciseIceClip`
Renamed `h_canNonTrivialCeilingClip` to `h_canIceClip`
